### ISSUE TITLE
[feature request] link file relative to wiki index

### ISSUE THREAD

**Issue by paperbenni** [vimwiki/vimwiki#1154]
it would be really useful if there was a way to link ```file:something``` relative to the wiki index file. Absolute paths don't work for me because I keep my wiki in different places on different devices and relative paths cannot be copied between different wiki files and obviously break when moving the wiki entry they're in. Linking relative to the wiki root/index would be really useful for a lot of use cases, off the top of my head one solution would be to introduce a prefix like ```~``` but for the vimwiki root directory

**Comment by brennen** [comment]
If I understand your needs correctly, this is already supported by using a leading `/`, which specifies an "absolute" path as far as the wiki is concerned (not an absolute path on the filesystem as a whole).  `[[/foo]]`links to `foo.wiki` at the root.

**Comment by paperbenni** [comment]
@brennen that's not true, file:/foo is relative to the entire file system. ```file:/foo``` looks in the root folder and I have to prefix it with /media/benjamin/hdd/wiki/ to get it working

**Comment by paperbenni** [comment]
And the link doesn't work on systems where I don't have the wiki cloned to a second harddrive

**Comment by brennen** [comment]
Ah, sorry, I glossed over the `file:` part and misunderstood your use case.

It seems like there's a reasonable case for this feature, though I'd like to be thoughtful about introducing yet more link syntax or a new arbitrary prefix. The link handling code is already pretty ugly/complex, and now that there's a [specification](https://github.com/vimwiki/vimwiki/blob/dev/doc/specification.wiki) and multiple implementations, we need to be deliberate about changes there.

In the meantime, you can probably get your desired behavior by writing a custom link handler - see `:help VimwikiLinkHandler`. Here's [an example](https://code.p1k3.com/gitea/brennen/bpb-kit/src/commit/1e938f505339f801eefd5ff43c6ba3ce46f7c2aa/home/.vim/vimrc#L378) I wrote to change how `file:` links work by default, and to add a `workspace:` namespace that I use for file storage.
